### PR TITLE
chore: calendar notes

### PR DESCRIPTION
**Pull request by HbDevelop**

### CODE REVIEW

File: projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts [hunk 1]
```diff
@@ -131,4 +138,20 @@ export class CalendarMonthCellComponent {
   trackByEventId = trackByEventId;
 
   validateDrag = isWithinThreshold;
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes?.notes?.currentValue) {
+      const notesOfTheDay = this.notes.filter(
+        (note) => note.start <= this.day.date && note.end >= this.day.date
+      );
+      if (!!notesOfTheDay.length) {
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts:147]: are those !! necessary with a number ?

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        >
+        </ng-template>
+      </div>
+    </div>
+  `,
+  styleUrls: ['./calendar-month-row.scss'],
+})
+export class CalendarMonthRowComponent implements OnChanges {
+  @Input() notes: CalendarEvent[];
+  @Input() view: any;
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:31]: typable ?

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+    if (d !== -1) {
+      left = d * (100 / 7);
+    } else {
+      left = 0;
+    }
+    return 'calc(' + left + '% + 3px)';
+  }
+
+  manageTop(notes: CalendarEvent[]) {
+    let tabWithoutOrder = [];
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:94]: type

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+          value.meta.left = this.manageLeft(value);
+          value.meta.width = this.manageWidth(value);
+        });
+        values.sort((a, b) => a.meta.order - b.meta.order);
+      });
+    }
+  }
+
+  manageLeft(note: CalendarEvent): string {
+    const d = this.daysSliced
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:81]: Can we use a more comprehensive name ?

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+  }
+
+  manageTop(notes: CalendarEvent[]) {
+    let tabWithoutOrder = [];
+    let ordersReserved: number[] = [];
+
+    tabWithoutOrder = notes
+      .filter((value) => !value.meta.order)
+      .sort(
+        (a, b) =>
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:100]: names

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+          this.dateDiffIndays(a.end, a.start) -
+          this.dateDiffIndays(b.end, b.start)
+      );
+    ordersReserved = notes
+      .filter((value) => value.meta.order)
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:105]: name

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+      .map((value) => +value.meta.order);
+
+    tabWithoutOrder.forEach((v) => {
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:108]: names v & i

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+    const indexEndDate = this.daysSliced
+      .map((daySliced) => daySliced.date)
+      .findIndex((date) => date.valueOf() === note.end.valueOf());
+    // start et end dans le row en cours
+    if (indexEndDate !== -1 && indexStartDate !== -1) {
+      width = (indexEndDate - indexStartDate + 1) * (100 / 7);
+    }
+    // start dans le row en cours et pas le end
+    else if (indexStartDate !== -1 && indexEndDate === -1) {
+      const a = this.dateDiffIndays(note.start, this.daysSliced[6].date);
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:133]: name

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+      width = (a + 1) * (100 / 7);
+    }
+    // end dans le row en cours et pas le start
+    else if (indexStartDate === -1 && indexEndDate !== -1) {
+      const a = this.dateDiffIndays(this.daysSliced[0].date, note.end);
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:138]: name

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+
+  dateDiffIndays(date1: Date, date2: Date): number {
+    return Math.floor(
+      (Date.UTC(date2.getFullYear(), date2.getMonth(), date2.getDate()) -
+        Date.UTC(date1.getFullYear(), date1.getMonth(), date1.getDate())) /
+        (1000 * 60 * 60 * 24)
+    );
+  }
+
+  hideSpan(note, day): boolean {
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:156]: types off note and day

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 11]
```diff
@@ -0,0 +1,202 @@
...
+    return day.date < note.start || day.date > note.end;
+  }
+
+  deepCopyFunction(obj) {
+    var rv;
```
Comment (Rigerwal) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:161]: name ?

**Review by Rigerwal — CHANGES_REQUESTED** [review]

File: projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts (hunk 1 above)
Comment (HbDevelop) [reply projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts:147]: not necessary while if(0) is falsy

**Review by DKarim — APPROVED** [review]

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 12]
```diff
@@ -0,0 +1,205 @@
...
+  @Input() rowIndex: number;
+  @Input() cellMonthNoteTemplate: TemplateRef<any>;
+
+  startDate: Date;
+  endDate: Date;
+
+  daysSliced: WeekDay[] = [];
+  firstDayOfRow: WeekDay;
+
+  concernedNotes: CalendarEvent[];
```
Comment (Njol75) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:41]: i'm not a big fan on this naming, i'm not sure what you want to say with "concerned" ?

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts [hunk 13]
```diff
@@ -0,0 +1,205 @@
...
+
+  notePerDay: Map<WeekDay, CalendarEvent[]> = new Map();
```
Comment (Njol75) [inline projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:43]: maybe `notesPerDay`

File: projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts [hunk 14]
```diff
@@ -131,4 +138,20 @@ export class CalendarMonthCellComponent {
   trackByEventId = trackByEventId;
 
   validateDrag = isWithinThreshold;
+
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes?.notes?.currentValue) {
+      const notesOfTheDay = this.notes.filter(
```
Comment (Njol75) [inline projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts:144]: `dayNotes` ?

File: projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts [hunk 15]
```diff
@@ -131,4 +138,20 @@ export class CalendarMonthCellComponent {
...
+        (note) => note.start <= this.day.date && note.end >= this.day.date
+      );
+      if (notesOfTheDay.length) {
+        this.marginTop =
```
Comment (Njol75) [inline projects/angular-calendar/src/modules/month/calendar-month-cell.component.ts:148]: maybe add a comment here or put it in function with a meaningful name, wdyt?

**Review by Njol75 — CHANGES_REQUESTED** [review]

File: projects/angular-calendar/src/modules/month/calendar-month-row.component.ts (hunk 12 above)
Comment (HbDevelop) [reply projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:41]: notesOfCurrentRow ?

Comment (Njol75) [reply projects/angular-calendar/src/modules/month/calendar-month-row.component.ts:41]: i'd prefer `currentRowNotes` (i'm not a big fan of using 'of' in variables name)

**Review by Njol75 — APPROVED** [review]

**Pull request merged**